### PR TITLE
Add convenience methods to sign and verify w/ RSA

### PR DESCRIPTION
**Pull request by cmurphy**
This patch adds wrapper methods to allow the user to sign and verify a
single message block without having to go through the multi-step
process of creating a signer or verifier, updating it with the one
message, and finalizing the result. This will make signing and
verifying data more user-friendly when only using small messages.

Partial bug #1529

### CODE REVIEW

**Comment by cmurphy** [comment]
Needs tests and docs

**Comment by alex** [comment]
jenkins, ok to test

**Comment by reaperhulk** [comment]
jenkins, retest this please

File: docs/hazmat/primitives/asymmetric/rsa.rst [hunk 1]
```diff
@@ -486,7 +514,8 @@ Key interfaces
 
         .. versionadded:: 0.3
 
-        Sign data which can be verified later by others using the public key.
+        Get object with which to sign potentially multiple blocks of data
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/rsa.rst]: The wording here is confusing. Could you rephrase it a bit?

File: docs/hazmat/primitives/asymmetric/rsa.rst [hunk 2]
```diff
@@ -580,8 +628,8 @@ Key interfaces
 
         .. versionadded:: 0.3
 
-        Verify data was signed by the private key associated with this public
-        key.
+        Get object with which to verify that potentially multiple blocks of
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/rsa.rst]: (Same as above)

**Comment by cmurphy** [comment]
@reaperhulk thanks for those notes, I've reworded it to hopefully be simpler.

**Comment by reaperhulk** [comment]
Thanks for the PR @cmurphy! Feel free to add yourself to the AUTHORS file in a new PR if you'd like.
